### PR TITLE
Reorganize folders of DataScienceTutorials.jl

### PR DESCRIPTION
**Pull request by EssamWisam**
This includes removing useless files, reorganizing common ones together, centralizing the routing index at the top-level and most importantly using a folder layout instead of prefixes to distinguish different tutorial types.

### CODE REVIEW

File: README.md
Comment (EssamWisam) [inline README.md:60]: This refers to the old `index.md`. The referred to part is now in `how-to-run-code` hence the change.

**Comment by EssamWisam** [comment]
image

Why does this heading exist? `Old Instructions` gives the impression that there are "alternative instructions"; meanwhile, in reality they do add information about exactly three things :
- How to visualize after changing (e.g., adding a new tutorial)
- Adding plots 
- Troubleshooting

I would vote for the following:
- Get rid of the potentially misleading "Old Instructions" title
- So far, I added briefly how to visualize the changes as the last step in the "add tutorial section" instead I can link to here (which is less brief and more informative) or get rid of this section.
- The "Adding Plots" section is not absolutely necessary because it was mentioned earlier to follow the blueprint by copying another tutorial which makes that easy. Thus, we can either remove this section or keep it and link to it in the "adding tutorials section. In the latter case, we have to remove what's mentioned there on `PyPlot`
- Troubleshooting section should stay.

**Comment by ablaom** [comment]
@EssamWisam I'm very happy for you to make an executive decision about these details. Feel free to rewrite these instructions as much as you like. 

Since the original developer has moved on, just go ahead do your best. You are the active maintainer with the most knowledge about the how the process now works and I will defer to you on this.  

Do ping me when you are ready for final review.

**Comment by EssamWisam** [comment]
Alright. Ready for review. Here is a [pretty print](https://github.com/JuliaAI/DataScienceTutorials.jl/blob/77aeee5f6a01bbe0d73ed73d1b53309827c1bf42/README.md) of the new README for easier review. The rest as can be seen are folder organization changes. I tested rendering the website after I did those.

File: README.md
Comment (ablaom) [inline README.md:45]:
```suggestion
```

The Manifest.toml is no longer tracked, which is good.

File: README.md
Comment (ablaom) [inline README.md:45]:
```suggestion
└── Project.toml         # environment for dev operations, eg. deploy.jl
```

File: README.md
Comment (ablaom) [inline README.md:52]: This internal link (and another like it) do not seem to work, at least in the preview.

**Review by ablaom — APPROVED** [review]
This is looks great, thanks. Just the few minor points.

File: README.md
Comment (EssamWisam) [reply README.md:52]: Well, I followed [this](https://stackoverflow.com/questions/27981247/github-markdown-same-page-link) which claims it works on Github and it does work locally for me. Let's merge and if it doesn't work I will make rightaway make a PR when I see it to replace it with "as in the... section below" or similar.

**Pull request merged**
